src/main/java/org/craftercms/engine/store/s3/S3Context.java

```diff
@@ -46,14 +46,14 @@ public S3Context(final String id, final ContentStoreAdapter storeAdapter, final
      * Returns the name of the bucket.
      */
     public String getBucket() {
-        return rootFolderUri.getBucket();
+        return rootFolderUri.bucket().orElseThrow();
```

Comment:
Let's specify the bucket name in the exception thrown message

Reply:
It only throws when the bucket is null so we won't have the bucket name:
```
    /**
     * Returns the bucket specified in the URI. Returns an empty optional if no bucket is specified.
     */
    public Optional<String> bucket() {
        return Optional.ofNullable(bucket);
    }
```

 I updated to use a new `S3BucketNotFoundException` and created a test case.